src/sys/wait.rs

```diff
@@ -112,149 +96,52 @@ impl WaitStatus {
     }
 }
 
-#[cfg(any(target_os = "linux",
-          target_os = "android"))]
 mod status {
     use sys::signal::Signal;
-    use libc::c_int;
-    use libc::SIGTRAP;
+    use libc::{self, c_int, SIGTRAP};
 
     pub fn exited(status: i32) -> bool {
-        (status & 0x7F) == 0
+        unsafe { libc::WIFEXITED(status) }
```

Comment:
They're marked as unsafe in `libc` actually. They're actually also marked as `extern "C"` which is also interesting. I definitely don't think they need to be either of those. @alexcrichton do you know why this is?